### PR TITLE
More listType validations

### PR DESCRIPTION
**Pull request by seans3**

### CODE REVIEW

File: pkg/generators/extension_test.go [hunk 1]
```diff
@@ -424,29 +449,81 @@ func TestValidateMemberExtensions(t *testing.T) {
...
 	}{
-		// Test single member extension
+		// List type extension must be defined on a slice.
 		{
-			extensions: []extension{patchStrategyExtension},
+			extensions: []extension{listTypeAtomicExtension},
 			member:     mapField,
 		},
-		// Test multiple member extensions
+		// Map keys defined, but missing listType=map
```
Comment (apelisse) [inline pkg/generators/extension_test.go]: There are probably many other use-cases here that are missing. You could almost create a more thorough list, and then use TDD, and maybe comment out those that are not implemented yet.

File: pkg/generators/extension.go [hunk 2]
```diff
@@ -34,25 +34,38 @@ type extensionAttributes struct {
 	allowedValues sets.String
 }
 
+const LIST_TYPE = "listType"
+const LIST_MAP_KEY = "listMapKey"
```
Comment (mbohlool) [inline pkg/generators/extension.go]: Should these names be camel-case?

Comment (mbohlool) [reply pkg/generators/extension.go]: Also do they need to be public? (i.e. start with capital letter)

**Comment by apelisse** [comment]
@seans3 @lavalamp Are we dropping this?

**Pull request closed without merging**